### PR TITLE
Malcolm in the Middle

### PR DESCRIPTION
**Pull request by GreatBigFailure**
This is a [wip]. I haven't even attempted to compile and plenty of bits aren't hooked up to each other.

I was asked to make a Malcolm in the Middle gamemode. I've only seen a few episodes and don't remember anything about it. So I read about 5 random paragraphs from the wiki then proceeded to make the rest up. This is the perfect Halloween content.

Sprites are by Bomberharris and myself. The better ones are Harris.
Shiftyrail did the actual gamemode part of the code, as in the stuff that makes any of it accessible. 

**_This whole thing is obviously bus only and should stay bus only forever and ever._**

### What

The crew is split into 3 factions: Hal, Lois, and the brothers. Each faction gets points each tick based on the location of malcolm:
Hal gets points for the west
Lois gets points for the east
Malcolm gets points for the middle

Malcolm cannot move in most cases and must be dragged or otherwise transported by his parents or brothers. He can perform any other action normally.

### The Brothers

The brothers seem out matched, with a smaller amount of territory and numbers against them but the crew which inherit the fragments of Dewey, Malcolm, and Reese get mutated by Malcolm's solipsism. 

**Dewey**
- Smaller and faster than others
- When shot at Dewey's survival instincts kick in and he'll quickly generate a sentient polyp asexually. The polyp is a much less durable or intelligent version of himself. 
- Polyps will consume the held items of those they attack
- Dewey can make all those around him feel extremely nauseous, causing mass vomiting
- Dewey will not be denied

dewey

**Malcolm**
- Malcolm cannot freely move unless he pauses time to speak to the audience, which I assume is probably something that happens in the show. This is not something he can manifest easily due to the self imposed bindings he has put on himself.
- Malcolm, being in the middle of all things, can easily revive his brothers by just believing they never died. His self imposed mortal bindings make the process of convincing himself take some time, however.
- Other stuff goes here when I think it up

**Reese**
- Reese once punched a nerd so hard it lifted him off the ground. Reese can punch walls and other structures to destroy them.
- Reese is the eldest brother and can consume Dewey polyps to fuel himself
- As the biggest and toughest Reese is seen as the school bully by Malcolm. He can tap into Malcolm's solipsism using that and generate lockers to stuff people into
- Reese can, at a distance, pick up an entire spaceman and hold him in his hands like a mouse.
- Reese (probably) tried out for the football team in season something episode maybe and was likely really good at it! So his tackles are absolutely terrifying
- Lois says that Reese caused his own premature birth by kicking her uterus endlessly while she was pregnant. Reese maintains the ability to provide phantom infant kicks even in his new form

reese2

:cl:
 * rscadd: Malcolm in the Middle Gamemode

### CODE REVIEW

**Comment by ValkyrieSkies** [comment]
(Francis is the eldest brother)

**Comment by GreatBigFailure** [comment]
Wait who the fuck is reese

**Comment by ValkyrieSkies** [comment]
Francis is the eldest brother who's off in military school

**Comment by GreatBigFailure** [comment]
Well then how is it not "Malcolm AND Reese in the Middle but like, less Middle and more just In Between Francies and Dewey" instead of "Malcolm in the Middle"

Fuck this Francis never existed. Malcolm erased him from the timeline. Deep lore.

**Comment by ValkyrieSkies** [comment]
Whoa...

**Comment by Zth--** [comment]
Bless those codersprites

**Comment by GreatBigFailure** [comment]
> Bless those codersprites

It's really very embarrassing how long they took me.

**Comment by d3athrow** [comment]
honestly terrifying

**Comment by Eneocho** [comment]
YES NO MAYBE I DONT KNOW

**Comment by Servalfang** [comment]
i dont care that its a draft speedmerge NOW

**Comment by Eneocho** [comment]
> 
> 
> Well then how is it not "Malcolm AND Reese in the Middle but like, less Middle and more just In Between Francies and Dewey" instead of "Malcolm in the Middle".

He's in the middle because Lois then had Jamie, which placed Malcolm straight in the middle. 
In order of age, Jamie, Dewey, Malcolm, Reese and Francis.

There's also that implied 6th child that's mentioned on the final episode, but that one doesn't count.

For Dewey, instead of a polyp, make it a hamster inside a hamsterball, since in S3E6, he releases Bernard (the class hamster) into the wild, which becomes a running gag on the show, the hamster eventually makes it all the way from California to Alaska.
Dewey also was a musical prodigy, he made a makeshift pipe organ on S5E16 after Hal refused to buy him a piano. He should have some piano/organ notes playing in the background whenever he approaches.

For Malcolm, he doesn't pause time per se. Sometimes events keep happening after he begins the 4th wall break, sometimes everything stands "still" while he explains. See the cold one for S1E3 for a good example of the latter.
Malcolm had a very strong determination. When Lionel Herkabe appeared on S3E2 and started pressuring the Krelboynes into a scoring system, Malcolm, who hated the toxic behaviour the Krelboynes took in an effort to please the teacher, pushed and pushed further into said scoring system, eventually making every krelyboyne but him crack under the pressure. Malcolm should be the toughest of them all, by far, and/or make any damage he recieves be somewhat sent back to the source (1/5th pain mirror?).

Reese was a really good cook, to the point of him choosing to obey Lois and Hal if punished with NOT cooking. He could produce random hard to make foods when near family, if the family eats it, they heal. Reese was not the strongest bully in school, so no tackling buff, but was the most feared due to technique and style. He gets his ass kicked in S1E8 by a Krelboyne's bigger brother and a girl in a wrestling match in S2E10. However, he singlehandedly brings back "order" to the school by returning as a bully after he stepped down and a whole lot of lesser bullies tried to take their place, again, see S2E10.

He also didn't try for a football team, but rather for the cheerleader team in S1E12, since the male side of the family had a "curse" that made the males act like retards whenever they were in front of someone they liked. Perhaps he should be able to buff the other brothers?

The brothers should also have a Nuclear option (Mentioned in the last episode, it's the evidence for the worst thing they did as brothers and reason they are nominally friendly with each other instead of backstabby. They made Lois think she had cancer). If the Nuclear option is used, the brothers/lois gets some mad buffs.

t. Malcolm in the Middle Academician

**Comment by jwhitak** [comment]
this PR is a fever dream come true

**Comment by robintime** [comment]
jesus christ

**Comment by cloutcode** [comment]
4 dreams a week i get shot in

**Comment by Eneocho** [comment]
> 
> 
> 4 dreams a week i get shot in

Requesting this as an actual line said by malcolm.

**Comment by SonixApache** [comment]
I'm surprised you translated 'malcolm in the middle mode' into this instead of it's actual mode.

**Comment by DamianX** [comment]
huh?

**Comment by ghost** [comment]
> YES NO MAYBE I DONT KNOW

can you repeat the question?

**Comment by Kromkar** [comment]
> > YES NO MAYBE I DONT KNOW
> 
> can you repeat the question?

YOU'RE NOT THE BOSS OF ME NOW

**Comment by Eneocho** [comment]
> > > YES NO MAYBE I DONT KNOW
> > 
> > 
> > can you repeat the question?
> 
> YOU'RE NOT THE BOSS OF ME NOW

YOU'RE NOT THE BOSS OF ME NOW

**Comment by beesil** [comment]
what the fuck

**Comment by GreatBigFailure** [comment]
I accidentally made all the spell descriptions fan fiction excerpts that only very, very vaguely hint at what the spell does.
I don't know if I should change that or what

**Comment by Eneocho** [comment]
> 
> 
> I accidentally made all the spell descriptions fan fiction excerpts that only very, very vaguely hint at what the spell does. I don't know if I should change that or what

I'd keep those as flavourtext but add a small line underneath that actually tells you the effects. Also, consider my MITM post when adding features.

**Comment by SonixApache** [comment]
i applaud your dedication but
what the actual fuck are you doing

**Review by Kurfursten — CHANGES_REQUESTED** [review]
You accidentally reverted some cult antag hud icons

**Comment by Eneocho** [comment]
> 
> 
> You accidentally reverted some cult antag hud icons

What the fuck based

**Comment by Blithering** [comment]
Need to code in a Francis who cannot spawn ever due to being in military school. He would be the largest and most respected offspring, but not able to exist.

**Comment by KingofHONK** [comment]
What the fuck is this?

**Comment by Eneocho** [comment]
> 
> 
> What the fuck is this?

It's beautiful, that's what it is.

**Comment by Eneocho** [comment]
> 
> 
> Need to code in a Francis who cannot spawn ever due to being in military school. He would be the largest and most respected offspring, but not able to exist.

He should arrive 10 minutes later since he emancipates and leaves military school, thus coming to the station.

**Comment by drawsstuff** [comment]
We're living in a DINOWARS SOON timeline

**Comment by GreatBigFailure** [comment]
This is still very alive. I'm just very very busy with non-spess things lately.
Will return to this soon, probably with the turning of the year. Something something year of The Middle.

**Comment by Eneocho** [comment]
Remember to add Francis as the latejoin brother (10 to 15 minutes in).

**Comment by GreatBigFailure** [comment]
> Remember to add Francis as the latejoin brother (10 to 15 minutes in).

Francis does not exist. Malcolm created Francis as a mental block distancing himself from The Middle. If Francis exists then Malcolm is no longer the middle child and therefor cannot be the middle of all things. When Malcolm realized Francis is a figment of his own doubts he awoke to the fact that he is the God Seed of the universe and the middle of all things. He realized that all of existence is a fictional manifestation of his own subconscious beliefs of what the universe should be.
This idea was outlandish enough that he continued to doubt it even as reality warped to match the realization, and so the Bindings of Francis chained themselves to him, restricting his solipsistic powers over reality. The game mode will be an exploration of Malcolm overcoming these doubts and erasing the last remnant of Francis, undoing his bindings and ushering in the Age of The Middle. An age that has always been as time itself will bend and reconstruct itself to match Malcolm's vision.

**Comment by Kurfursten** [comment]
As per your latest commit, is this a trojan horse to sneak in angler? Because that won't work, I do read the commits

**Comment by GreatBigFailure** [comment]
> As per your latest commit, is this a trojan horse to sneak in angler? Because that won't work, I do read the commits

Copy pasting my explanation:

I still can't use git. However when I first started I was even worse and didn't understand branches. Long story short is I've been waiting until angler was finished to clean up my working tree because I was too scared I'd accidentally delete everything. This has finally come back to bite me

**Comment by Eneocho** [comment]
You should've finished the fishing PR, Rubylips. The fishing metaclub demands their sacrifice.

**Comment by Kurfursten** [comment]
I'll take the tag of shame off when you disentangle your PR from the angler's net

**Comment by Blithering** [comment]
Malcolm has been waylaid by fish. The fish demand their tithe.

**Comment by Eneocho** [comment]
I'd honestly be happy with either of the PRs going through.
I've been waiting for that fishing thing for more than a year now. 
And malcolm in the middle is SOVLFVL.

**Comment by GreatBigFailure** [comment]
You'll get both eventually.
Malcolm is almost done, I'm just busy. Angler is, as you can see, a fucking lot of content and will still be a bit. Current plan is Malcolm, then a small update to tackling, then back to angler. Right now angler is mostly held back by how disorganized it is, I didn't document what parts have small unfinished things very well so I need to do that. I need to do some tedious file sorting like actually putting a lot of dmis in place etc, then ill go through everything for testing and what not. A lot of this can be done at the same time

**Comment by Eneocho** [comment]
Do you plan on adding Francis (as a player or in game DEEP LORE(TM)) or any other thing (like my suggestions) to malcolm?

**Comment by GreatBigFailure** [comment]
> Do you plan on adding Francis (as a player or in game DEEP LORE(TM)) or any other thing (like my suggestions) to malcolm?

Using it as some inspiration for the spells I still need to add to make the malcolm role not boring for the player in it. I'm considering an "Invoke Francis" spell that does -something- major but costs the brother team a significant amount of points due to DEEP LORE. Probably not though because that's more complicated and time consuming to do, and this was meant to be done over a month ago.

**Comment by GreatBigFailure** [comment]
Shiftyrail donated about an hour of his time to code what was taking me a week, which is a potentially informative conversion rate.
Anyway, he coded all the actual gamemode bits. The stuff that makes it possible to actually happen, prevents 20 wizards from ruining the bus event, and prevents malcolm from triggering when not summoned.

File: code/datums/gamemode/factions/wilkersons/wilkerson_spells.dm [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+/spell/deweyVomit/cast(var/list/targets, var/mob/user)
+	for(var/mob/living/carbon/human/H in view(9, user))
+		H.vomit()
+		H.Jitter(30)
+		H.stuttering += 30
+
+/spell/deweyPolyp
+	name = "Split"
+	desc = "Create a mindless copy of yourself."
+	deepLore = "Dewey was always around. Malcolm had to rationize it, and so it was as he perceived."
```
Comment (Eneocho) [inline code/datums/gamemode/factions/wilkersons/wilkerson_spells.dm:132]: Wouldn't it be "rationalize" instead?

File: code/datums/gamemode/factions/wilkersons/wilkerson_spells.dm [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	charge_max = 50
+
+/spell/deweyPolyp/cast(var/list/targets, var/mob/user)
+	var/mob/living/carbon/monkey/dewey/D = user
+	D.visible_message("<span class='warning'>[D] grows dislodges a chunk of flesh from himself!</span>")
+	D.splitDewey()
+
+/spell/deweyEnrage
+	name = "Forego Medication"
+	desc = "All Dewey's will enter a curious fervor. They will latch on to their victims and jump them back to your side."
```
Comment (Eneocho) [inline code/datums/gamemode/factions/wilkersons/wilkerson_spells.dm:142]: It should be All Deweys, meaning all the Dewey individuals, not All Dewey's.

File: code/modules/admin/verbs/tagmode.dm [hunk 3]
```diff
@@ -6,12 +6,30 @@
...
+		return
+
+	if (ticker.malcolm_mode_enabled)
+		var/confirm_cancel = alert(user, "Do you wish to cancel malcolm mode?", "Malcom in the Middle!", "Yes", "No")
+		if (confirm_cancel == "Yes")
+			message_admins("[key_name(user)] has cancelled malcolm mode.")
+			log_admin("[key_name(user)] has cancelled malcolm mode.")
+			ticker.cancel_malcolm_middle(user)
+	else
+		var/confirm = alert(user, "Do you wish to enable malcolm mode?", "Malcom in the Middle!", "Yes", "No")
```
Comment (Eneocho) [inline code/modules/admin/verbs/tagmode.dm:31]: Typo on Malcolm's name here.

**Comment by Eneocho** [comment]
What would the "You're not the Boss of Me Now" spell do? There is currently nothing there.

**Comment by GreatBigFailure** [comment]
> What would the "You're not the Boss of Me Now" spell do? There is currently nothing there.

I haven't actually decided yet. Since Malcolm is mostly immobile and both a player and effectively the flag in capture the flag his options need to be based on that. What he has so far is mostly to circumvent his immobility when he needs to. I don't want it to be too annoying to drag Malcolm around though, and I don't want to give him too many reset buttons.
So something like a simple projectile doesn't sound good because you need to drag him around, so you're just going to die immediately since he's going to land it no problem. 
If you have any ideas I'm all ears.

Also thanks for the typo fixes.

**Comment by Eneocho** [comment]
Some ideas:
A long time to charge mapwide time freeze that lasts 30 seconds, while everyone's cameraport pans to malcolm (so they can hear his monologue). He can move during that time.
An expensive skill (long time to charge and damages everything on a 5x5 area surrounding faux-malcolm, including real-malcolm) that temporarily turns malcolm into a "human", crippling his solipsism (no revivals if the family dies during this skill), leaving a duplicate behind while the real malcolm moves away from the middle (he can now walk). 
One use access to a single powerful item of malcolm's choice (from meteor gun to nuke decoder).
Malcolm demands obedience from reality and shuts down all of X thing for 5 minutes (all guns, all power sources, all medicine, etc).
Malcolm rebels against the code itself and shuts down an entire code subsystem (at random) for 5 minutes. This could do shit from temporarily nuking the electrical system to removing damage altogether.
He rebels against reality itself, returning the entire station (sans flag capture points) to roundstart (engines and medicine get undone, station damage gets reverted, all dead players get restored to their bodies (if applicable)).
Francis, being the prime rebel, was the first one to defy the will of the Mother. He was banished from reality when Dewey was born as to keep Malcolm in the Middle. Malcolm can summon Francis, the non-existent, who then starts to consume reality itself, trying to achieve existence (spawns a temporary non-expanding bluespace sea on a random tile in a 5x5 tile radius around Francis).
Severely damages a body part on anyone with authority (we define authority here as having a firearm). This means that both sec, command and the tide would get very hurt, damaging all antagonistic parties.
Malcolm rebels against the admins themselves, gaining access to the spawn menu for 1 minute.

**Comment by GreatBigFailure** [comment]
> Malcolm rebels against the admins themselves, gaining access to the spawn menu for 1 minute.

Some of the other stuff doesn't match the mode or would just be way too hard to code. This though, this I like. Thanks.

**Comment by Eneocho** [comment]
> > Malcolm rebels against the admins themselves, gaining access to the spawn menu for 1 minute.
> 
> Some of the other stuff doesn't match the mode or would just be way too hard to code. This though, this I like. Thanks.

If you do implement it, make sure to add an ominous message to everyone in the world.
Something like "The Gods themselves are taken aback, the Middle is almost complete..."

**Comment by GreatBigFailure** [comment]
I added a mechanic that announces a deep lore message to an adjustable percentage of players when one of the brothers uses a spell. It's to compromise actually describing the spells with my accidental fanfiction descriptors. So the ominous message is taken care of.

**Comment by GreatBigFailure** [comment]
Nevermind, it's dead. Stopped being fun and I don't want to finish. Take what you want from what's here if someone wants it.

**Comment by Eneocho** [comment]
Life is unfaaaaaaaair

**Comment by Servalfang** [comment]
bring it back

**Pull request closed without merging**